lib/code_corps/stripe/adapters/stripe_card.ex

```diff
@@ -0,0 +1,9 @@
+defmodule CodeCorps.Stripe.Adapters.StripeCard do
+  import CodeCorps.MapUtils, only: [rename: 3]
+
+  def params_from_stripe(%{} = stripe_map) do
+    stripe_map
+    |> rename("id", "id_from_stripe")
+    |> rename("customer", "stripe_customer_id")
```

Comment:
On the stripe API model, `customer` is basically a foreign key, that can be null.

With my initial approach, were we simply used the original stripe object ids as string primary keys in our own models, this was just a matter of renaming `customer`, to `stripe_customer_id`, since it was an Id and we had a `belongs_to :stripe_customer` on a card model.

Now that we store stripe ids as separate fields, if we want to keep the relationship and do it somewhat straightforward, I could do one of several approaches

* Enforce a foreign key constraint from this field, into the `id_from_stripe` field on the `stripe_customers` table. I believe it is possible with ecto and postgres and should not be too difficult. We can also further rename this `customer` field into `customer_id_from_stripe` or something, to make it a bit cleaer.

* Keep the default `belongs_to :stripe_customer` and rewrite our changeset for stripe card to fetch the stripe customer by `id_from_stripe` and then assign it to the belongs_to.

* We could also just have both `stripe_customers` and `stripe_cards` simply belong to a user or an organization, the way a slugged route does . That might even make the most sense, since in the stripe API, a stripe card can belong either to a stripe customer, which belongs to a user, or to a stripe account, which belongs to a organization.

In any case, the field rename as is cannot stay around, so I'll remove it for now.